### ISSUE TITLE
Environment with nbconvert 5.6.1 and nbconvert-core 6.4.5

### ISSUE THREAD

**Issue by maximlt** [conda-forge/nbconvert-feedstock#73]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

I'm opening this issue after one of our Github action worflows started to fail. I noticed the conda environment it created ended up with `nbconvert-5.6.1` and `nbconvert-core=6.4.5`. Having nbconvert 5 in that environment is expected, we have a dependency (nbsite) that pins it to <6. I can't explain why nbconvert-core ended up being installed to. This can sort of reproduced with this command, which should fail?

`conda create -n testo -c conda-forge -c nodefaults --override-channels python=3.8 "nbconvert<6" "nbconvert-core>=6"`

It seems like the actual version that gets installed is `6.4.5`.

```
>>> import nbconvert
>>> nbconvert.__version__
'6.4.5'
```

### Installed packages

```shell
# packages in environment at /Users/mliquet/miniconda3/envs/testo:
#
# Name                    Version                   Build  Channel
attrs                     21.4.0             pyhd8ed1ab_0    conda-forge
beautifulsoup4            4.10.0             pyha770c72_0    conda-forge
bleach                    4.1.0              pyhd8ed1ab_0    conda-forge
bzip2                     1.0.8                h0d85af4_4    conda-forge
ca-certificates           2021.10.8            h033912b_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
importlib-metadata        4.11.3           py38h50d1736_1    conda-forge
importlib_resources       5.6.0              pyhd8ed1ab_0    conda-forge
jinja2                    3.1.1              pyhd8ed1ab_0    conda-forge
jsonschema                4.4.0              pyhd8ed1ab_0    conda-forge
jupyter_client            7.2.1              pyhd8ed1ab_0    conda-forge
jupyter_core              4.9.2            py38h50d1736_0    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
libcxx                    13.0.1               hc203e6f_0    conda-forge
libffi                    3.4.2                h0d85af4_5    conda-forge
libsodium                 1.0.18               hbcb3906_1    conda-forge
libzlib                   1.2.11            h6c3fc93_1014    conda-forge
markupsafe                2.1.1            py38hed1de0f_1    conda-forge
mistune                   0.8.4           py38h96a0964_1005    conda-forge
nbclient                  0.5.13             pyhd8ed1ab_0    conda-forge
nbconvert                 5.6.1              pyhd8ed1ab_2    conda-forge
nbconvert-core            6.4.5              pyhd8ed1ab_1    conda-forge
nbformat                  5.2.0              pyhd8ed1ab_0    conda-forge
ncurses                   6.3                  he49afe7_0    conda-forge
nest-asyncio              1.5.4              pyhd8ed1ab_0    conda-forge
openssl                   3.0.2                h6c3fc93_1    conda-forge
packaging                 21.3               pyhd8ed1ab_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
pip                       22.0.4             pyhd8ed1ab_0    conda-forge
pygments                  2.11.2             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.7              pyhd8ed1ab_0    conda-forge
pyrsistent                0.18.1           py38h96a0964_0    conda-forge
python                    3.8.13          h66c20e1_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.8                      2_cp38    conda-forge
pyzmq                     22.3.0           py38hd3b92b6_1    conda-forge
readline                  8.1                  h05e3726_0    conda-forge
setuptools                61.3.0           py38h50d1736_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
soupsieve                 2.3.1              pyhd8ed1ab_0    conda-forge
sqlite                    3.37.1               hb516253_0    conda-forge
testpath                  0.6.0              pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h5dbffcc_0    conda-forge
tornado                   6.1              py38h96a0964_2    conda-forge
traitlets                 5.1.1              pyhd8ed1ab_0    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
xz                        5.2.5                haf1e3a3_1    conda-forge
zeromq                    4.3.4                he49afe7_1    conda-forge
zipp                      3.7.0              pyhd8ed1ab_1    conda-forge
zlib                      1.2.11            h6c3fc93_1014    conda-forge
```

### Environment info

```shell
active environment : testo
    active env location : /Users/mliquet/miniconda3/envs/testo
            shell level : 1
       user config file : /Users/mliquet/.condarc
 populated config files : /Users/mliquet/.condarc
          conda version : 4.12.0
    conda-build version : not installed
         python version : 3.9.5.final.0
       virtual packages : __osx=10.16=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /Users/mliquet/miniconda3  (writable)
      conda av data dir : /Users/mliquet/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/mliquet/miniconda3/pkgs
                          /Users/mliquet/.conda/pkgs
       envs directories : /Users/mliquet/miniconda3/envs
                          /Users/mliquet/.conda/envs
               platform : osx-64
             user-agent : conda/4.12.0 requests/2.27.1 CPython/3.9.5 Darwin/20.6.0 OSX/10.16
                UID:GID : 502:20
             netrc file : None
           offline mode : False
```

**Comment by minrk** [comment]
I think this will be fixed by #74

**Comment by bollwyvl** [comment]
Once the now-merged #74 is live on CDN, i'll open the admin-requests PR to mark the `1` as broken
